src/main/java/com/thebois/models/world/structures/StructureFactory.java

```diff
@@ -24,6 +31,7 @@ public static IStructure createStructure(
 
         return switch (type) {
             case HOUSE -> new House(position);
+            case STOCKPILE -> new Stockpile(position, inventory);
```

Comment:
There's a risk here that inventory is still null when this is called.

You should throw an exception if that is the case, take a look at how I did it for `RoleFactory`:
https://github.com/martinjonsson01/OOPP-WITH-THE-BOIS/blob/ed6c0f59c467c25b50e688d6f07874d48e8795e9/src/main/java/com/thebois/models/beings/roles/RoleFactory.java#L131-L145